atef/annotated_plans.py

```diff
@@ -0,0 +1,615 @@
+        return orig_func
+
+    return wrapper
+
+
+@parameter_annotation_decorator({
+    "parameters": {
+        "detectors": {"annotation": "typing.List[__DEVICE__]"},
+        "num": {"annotation": "int"},
+        "delay": {"annotation": "typing.Union[typing.Iterable[float], float]"},
+    }
+})
+@add_docstring_from(bp.count)
+def count(
+    detectors: List[Any],
+    num: int = 1,
+    delay: Union[Iterable[Union[float, int]], float, int] = None,
+    *,
+    per_shot: Optional[Callable] = None,
+    md: Optional[Dict[str, Any]] = None
+):
+    yield from bp.count(detectors, num, delay, per_shot=per_shot, md=md)
+
+
+@parameter_annotation_decorator({
+    "parameters": {
+        "detectors": {"annotation": "typing.List[__DEVICE__]"},
+        "motors": {"annotation": "typing.List[__DEVICE__]"},
+        "point_lists": {"annotation": "typing.List[typing.List[float]]"},
+    }
+})
+@add_docstring_from(bp.list_scan)
+def list_scan(
+    detectors: List[Any],
+    motors: List[Any],
+    point_lists: List[List[float]],
+    per_step: Optional[Callable] = None,
+    md: Optional[Dict[str, Any]] = None
+):
+    motor_args = [val for pair in zip(motors, point_lists) for val in pair]
+    yield from bp.list_scan(detectors, *motor_args, per_step=per_step, md=md)
+
+
+@parameter_annotation_decorator({
+    "parameters": {
+        "detectors": {"annotation": "typing.List[__DEVICE__]"},
+        "motors": {"annotation": "typing.List[__DEVICE__]"},
+        "point_lists": {"annotation": "typing.List[typing.List[float]]"},
+    }
+})
+@add_docstring_from(bp.rel_list_scan)
+def rel_list_scan(
+    detectors: List[Any],
+    motors: List[Any],
+    point_lists: List[List[float]],
+    per_step: Optional[Callable] = None,
+    md: Optional[Dict[str, Any]] = None
+):
+    motor_args = [val for pair in zip(motors, point_lists) for val in pair]
+    yield from bp.rel_list_scan(detectors, *motor_args, per_step=per_step, md=md)
+
+
+@parameter_annotation_decorator({
+    "parameters": {
+        "detectors": {"annotation": "typing.List[__DEVICE__]"},
+        "motors": {"annotation": "typing.List[__DEVICE__]"},
+        "point_lists": {"annotation": "typing.List[typing.List[float]]"},
+    }
+})
+@add_docstring_from(bp.list_grid_scan)
+def list_grid_scan(
+    detectors: List[Any],
+    motors: List[Any],
+    point_lists: List[List[float]],
+    snake_axes: bool = False,
+    per_step: Optional[Callable] = None,
+    md: Optional[Dict[str, Any]] = None
+):
+    motor_args = [val for pair in zip(motors, point_lists) for val in pair]
+    yield from bp.list_grid_scan(detectors, *motor_args, snake_axes,
+                                 per_step=per_step, md=md)
+
+
+@parameter_annotation_decorator({
+    "parameters": {
+        "detectors": {"annotation": "typing.List[__DEVICE__]"},
+        "motors": {"annotation": "typing.List[__DEVICE__]"},
+        "point_lists": {"annotation": "typing.List[typing.List[float]]"},
```

Comment:
Conversion of `*motor_args` into `list[motors]` and `list[point_lists]` is sensible to me 👍 

However the annotation "override" in the decorator appears to match the one in the type annotation of the plan. Is this line necessary - and if so, why?

Reply:
Ah it's not necessary, you're right.  I got caught autopiloting 😆